src/sql/src/optimizer_metrics.rs

```diff
@@ -66,7 +67,19 @@ impl OptimizerMetrics {
         self.e2e_optimization_time_seconds
             .with_label_values(&[object_type])
             .observe(duration.as_secs_f64());
-        if duration > self.e2e_optimization_time_seconds_log_threshold {
+        // Also log it when it's big.
+        let debug_threshold = soft_assertions_enabled();
```

Comment:
Wait, I think we also enable soft assertions in release mode in CI. Can‘t we check how the program was compiled?

Reply:
PR: https://github.com/MaterializeInc/materialize/pull/32841
Thanks for spotting this!